ddl/cluster.go

```diff
@@ -256,34 +249,38 @@ func (w *worker) onFlashbackCluster(d *ddlCtx, t *meta.Meta, job *model.Job) (ve
 			job.State = model.JobStateCancelled
 			return ver, errors.Trace(err)
 		}
+		job.SchemaState = model.StateNone
 		return ver, nil
 	} else if flashbackJobID != job.ID {
 		job.State = model.JobStateCancelled
 		return ver, errors.Errorf("Other flashback job(ID: %d) is running", job.ID)
 	}
 
+	sess, err := w.sessPool.get()
+	if err != nil {
+		job.State = model.JobStateCancelled
+		return ver, nil
+	}
+	defer w.sessPool.put(sess)
+
 	// Stage 2, check flashbackTS, close GC and PD schedule.
-	if job.SnapshotVer == 0 {
-		if err = checkAndSetFlashbackClusterInfo(w, d, t, job, flashbackTS); err != nil {
-			job.State = model.JobStateCancelled
-			return ver, errors.Trace(err)
-		}
-		snapVer, err := getValidCurrentVersion(d.store)
-		if err != nil {
+	if job.SchemaState == model.StateNone {
```

Comment:
If we are using `job.SchemaState` to distinguish the job stage, why not

```
switch job.SchemaState  {
case stage1:
...
default:
}
```

Reply:
Done, PTAL